mmf/datasets/multi_dataset_loader.py

```diff
@@ -186,9 +186,19 @@ def _infer_dataset_probabilities(self):
     def __len__(self):
         # Since, this is iterator, we need to return total length == number of batches
         batch_size = get_batch_size()
+        # Changed the length to accomadate drop_last == True
+        # drop_last is required if the batch is split into multiple cores
+        # some of the cores may not have enough examples.
+        if is_xla():
+            logging.info(
+                "drop_last is set to True to avoid uneven dimension shapes"
```

Comment:
```suggestion
                "drop_last is set to True to avoid uneven dimension shapes "
```

Reply:
incorporated in the next commit. Thanks.